src/main/java/org/javarosa/core/model/data/GeoTraceData.java

```diff
@@ -205,4 +207,16 @@ public Double toNumeric() {
     public String toString() {
         return getDisplayText();
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == null || getClass() != o.getClass()) return false;
```

Comment:
You could add a reference equality check as the first step:
`if (this == o) return true;
`the same in the other class.